File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/ShoppingList.java
package za.co.woolworths.financial.services.android.models.dto;

import com.google.gson.annotations.SerializedName;

public class ShoppingList {

    @SerializedName("id")
    public String listId;
    @SerializedName("name")
    public String listName;
    @SerializedName("itemCount")
    public int listCount;

    public boolean shoppingListRowWasSelected;
    public boolean wasSentToServer;
}

File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/ShoppingList.kt
package za.co.woolworths.financial.services.android.models.dto

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import za.co.woolworths.financial.services.android.shoppinglist.service.network.ProductListDetails

@Parcelize
data class ShoppingList(
    @SerializedName("id")
    var listId: String = "",

    @SerializedName("name")
    var listName: String = "",

    @SerializedName("itemCount")
    var listCount: Int = 0,
    var shoppingListRowWasSelected: Boolean = false,
    var wasSentToServer: Boolean = false,
    var modifiedListCount: String = "",
    var productImageList: List<ProductListDetails> = emptyList() // todo This will get replace once we get actual data from response.
): Parcelable